server/internal/milvus/milvus.go

```diff
@@ -102,10 +134,74 @@ func (s *S) DeleteVectorStore(ctx context.Context, name string) error {
 }
 
 // InsertDocuments inserts documents into a collection in milvus.
-func (s *S) InsertDocuments(ctx context.Context, name string, vectors [][]float32) error {
+func (s *S) InsertDocuments(ctx context.Context, name string, files, texts []string, vectors [][]float32) error {
 	vectorCol := entity.NewColumnFloatVector(vectorColName, len(vectors[0]), vectors)
-	if _, err := s.client.Insert(ctx, name, "" /* partitionName */, vectorCol); err != nil {
+	fileCol := entity.NewColumnVarChar(fileIDColName, files)
+	textCol := entity.NewColumnVarChar(textColName, texts)
+	if _, err := s.client.Insert(ctx, name, "" /* partitionName */, vectorCol, fileCol, textCol); err != nil {
 		return err
 	}
 	return nil
 }
+
+// DeleteDocuments deletes documents from a collection in milvus by fileID.
+func (s *S) DeleteDocuments(ctx context.Context, collectionName, fileID string) error {
+	if err := s.client.LoadCollection(ctx, collectionName, false); err != nil {
+		return fmt.Errorf("load collection: %s", err)
+	}
+	defer func() {
+		if err := s.client.ReleaseCollection(ctx, collectionName); err != nil {
+			log.Printf("release collection: %s\n", err)
+		}
+	}()
+
+	expr := fmt.Sprintf("%s like \"%s\"", fileIDColName, fileID)
+	return s.client.Delete(ctx, collectionName, "" /* partitionName */, expr)
+}
+
+// Search searches for the documents with similar vectors in milvus.
+func (s *S) Search(ctx context.Context, collectionName string, vectors []float32, numDocuments int) ([]string, error) {
+	if err := s.client.LoadCollection(ctx, collectionName, false); err != nil {
+		return nil, fmt.Errorf("load collection: %s", err)
+	}
+	defer func() {
+		if err := s.client.ReleaseCollection(ctx, collectionName); err != nil {
+			log.Printf("release collection: %s\n", err)
+		}
+	}()
+
+	sp, err := entity.NewIndexIvfFlatSearchParam(defaultIvfFlatSearchParam)
+	if err != nil {
+		return nil, err
+	}
+
+	vs := []entity.Vector{entity.FloatVector(vectors)}
+	results, err := s.client.Search(
+		ctx,
+		collectionName,
+		nil, /* partitions */
+		"",  /* expr */
+		[]string{primaryKeyColName, fileIDColName, textColName},
+		vs,
+		vectorColName,
+		defaultMetricType,
+		numDocuments,
+		sp,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []string
+	for _, r := range results {
+		if r.ResultCount == 0 {
```

Comment:
Could you add a comment to explain when `ResultCount` is zero?

Reply:
Not so sure, added a TODO